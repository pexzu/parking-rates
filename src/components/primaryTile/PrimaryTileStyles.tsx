import styled from '@emotion/styled';

export const PrimaryTileBlock = styled.div`
  label: PrimaryTileBlock;
  transition: 0.3s ease;
  display: flex;
  justify-content: space-between;
  padding: 15px;
  align-items: center;
  padding: 0 48px;
  box-shadow: 0 0 1px 0;
  background: #224172;
  cursor: pointer;
  :hover {
    opacity: 0.9;
  }
`;

export const PrimaryTileBlockLabel = styled.div`
  label: PrimaryTileBlockLabel;
  padding: 25px 0;
  font-size: 21px;
  color: #ffffff;
`;

interface IPrimaryTileBlockIconContainer {
  rotate: boolean;
}
export const PrimaryTileBlockIcon = styled.span<IPrimaryTileBlockIconContainer>`
  label: PrimaryTileBlockIcon;
  transform: ${props => (props.rotate ? 'rotate(180deg)' : 'rotate(0deg)')};
  transition: 0.3s ease;
  width: 24px;
  height: 24px;
  > svg {
    fill: #ffffff;
  }
`;
