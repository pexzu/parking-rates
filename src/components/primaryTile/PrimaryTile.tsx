import * as React from 'react';
import * as E from './PrimaryTileStyles';
import { DropdownIcon } from '../../assets/DropdownIcon';

export interface PrimaryTileProps {
  onClick: () => void;
  label: string;
  rotate: boolean;
}

export default class PrimaryTile extends React.PureComponent<PrimaryTileProps> {
  private tile = React.createRef<HTMLDivElement>();

  onClick = () => {
    this.props.onClick();
    console.log(this.tile.current);
    this.tile.current && window.scrollTo(0, this.tile.current.offsetTop);
  };
  render() {
    const { label, rotate } = this.props;
    return (
      <E.PrimaryTileBlock onClick={this.onClick} ref={this.tile}>
        <E.PrimaryTileBlockLabel>{label}</E.PrimaryTileBlockLabel>
        <E.PrimaryTileBlockIcon rotate={rotate}>
          <DropdownIcon />
        </E.PrimaryTileBlockIcon>
      </E.PrimaryTileBlock>
    );
  }
}
