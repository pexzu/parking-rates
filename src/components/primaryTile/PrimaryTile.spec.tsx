import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import { PrimaryTileBlock, PrimaryTileBlockIcon, PrimaryTileBlockLabel } from './PrimaryTileStyles';
import PrimaryTile from './PrimaryTile';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

describe('PrimaryTile component', () => {
  test('render', () => {
    const wrapper = shallow(<PrimaryTile rotate={true} label='testing' onClick={() => {}} />);
    expect(wrapper.exists()).toBe(true);
  });
});

//Snapshot Testing
it('PrimaryTileBlock renders correctly', () => {
  const tree = renderer.create(<PrimaryTileBlock />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('PrimaryTileBlockLabel renders correctly', () => {
  const tree = renderer.create(<PrimaryTileBlockLabel />).toJSON();
  expect(tree).toMatchSnapshot();
});
