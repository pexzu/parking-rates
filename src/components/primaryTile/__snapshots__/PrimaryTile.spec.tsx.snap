// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryTileBlock renders correctly 1`] = `
<div
  className="css-1vbjs70-PrimaryTileBlock"
/>
`;

exports[`PrimaryTileBlockLabel renders correctly 1`] = `
<div
  className="css-12zm5pv-PrimaryTileBlockLabel"
/>
`;
